lib/bson/document.rb

```diff
@@ -44,12 +44,15 @@ class Document < ::Hash
     #   document.fetch(:field)
     #
     # @param [ String, Symbol ] key The key to look up.
+    # @param [ Object ] default Returned value if key does not exist
+    # @yield [key] Block returning default value for given key
     #
     # @return [ Object ] The found value. Raises KeyError if none found.
     #
     # @since 4.4.0
-    def fetch(key)
-      super(convert_key(key))
+    def fetch(key, *args, &block)
+      key = convert_key(key)
```

Comment:
Why do we take `*args` here? `Hash#fetch` only takes a single optional argument after the key. I think the correct signature here would be  `def fetch(key, default = nil, &block)`.

Reply:
As written this is forward compatible should Ruby decide to add another argument to fetch.

Reply:
Sure, but it also makes it show up in the documentation as taking more arguments than can be passed to it without raising an error. I don't think that we should make any method we define that wraps a standard library one accept as many arguments as possible just in case it changes at some point.

Reply:
The behavior is also different when passing nil for default and not passing a default at all.

I updated the docstring following the suggestion in https://stackoverflow.com/questions/3934008/rdoc-darkfish-and-the-call-seq-tag#4628982.